pkg/controller/errors/errors.go

```diff
@@ -47,6 +47,31 @@ func IsMultipleExistingCRDOwnersError(err error) bool {
 	return false
 }
 
+// Declare the InstallPlanError type which is either retryable or not retry-able
+// in order to determine whether or not a subscription shoudld retry or fail
+type InstallPlanError struct {
+	message string
+}
+
+func (e InstallPlanError) Error() string {
+	return e.message
+}
+
+func NewInstallPlanError(message string) InstallPlanError {
+	return InstallPlanError{
+		message: message,
+	}
+}
+
+func IsRetryable(err error) bool {
+	switch err.(type) {
+	case InstallPlanError:
+		return false
+	}
+
+	return true
+}
+
```

Comment:
```suggestion
type RetryableError struct {
	error
}

func NewRetryableError(err error) RetryableError {
	return RetryableError{err}
}

func IsRetryable(err error) bool {
	switch err.(type) {
	case RetryableError:
		return true
	}

	return false
}
```

What do you think?

Reply:
Works for me. I had conceived of a solution where already existing error classes were divided into "retryable" and "unretryable" by a big switch statement in the sky but this is much cleaner. Will implement.